Modules/_pickle.c

```diff
@@ -3952,7 +3949,7 @@ save(PicklerObject *self, PyObject *obj, int pers_save)
              1   if a persistent id was saved.
          */
         if ((status = save_pers(self, obj)) != 0)
```

Comment:
Can this trigger a recursive call?

Reply:
`save_pers()` calls `save()` with `pers_save=1`, and this stops the recursion via save_pers.